src/System.Text.Json/tests/JsonObjectTests.cs

```diff
@@ -0,0 +1,609 @@
+            var addresses = person["addresses"] as JsonObject;
+            Assert.IsType<JsonObject>(addresses["office"]);
+            Assert.IsType<JsonObject>(addresses["home"]);
+        }
+
+        [Fact]
+        public static void TestAssignmentDefinition()
+        {
+            JsonNode employee = EmployeesDatabase.GetNextEmployee().Value;
+            Assert.IsType<JsonObject>(employee);
+        }
+
+        [Fact]
+        public static void TestAddingKeyValuePair()
+        {
+            var employees = new JsonObject
+            {
+                EmployeesDatabase.GetNextEmployee(),
+                EmployeesDatabase.GetNextEmployee(),
+                EmployeesDatabase.GetNextEmployee(),
+                EmployeesDatabase.GetNextEmployee(),
+            };
+
+            string prevId = "";
+            foreach ((string id, JsonNode employee) in employees)
+            {
+                Assert.NotEqual(prevId, id);
+                prevId = id;
+
+                Assert.IsType<JsonObject>(employee);
+            }
+        }
+
+        [Fact]
+        public static void TestAddingKeyValuePairAfterInitialization()
+        {
+            var employees = new JsonObject();
+            foreach (KeyValuePair<string, JsonNode> employee in EmployeesDatabase.GetTenBestEmployees())
+            {
+                employees.Add(employee);
+            }
+
+            string prevId = "";
+            foreach ((string id, JsonNode employee) in employees)
+            {
+                Assert.NotEqual(prevId, id);
+                prevId = id;
+
+                Assert.IsType<JsonObject>(employee);
+            }
+        }
+
+        [Fact]
+        public static void TestAddingKeyValuePairsCollection()
+        {
+            var employees = new JsonObject(EmployeesDatabase.GetTenBestEmployees());
+
+            string prevId = "";
+            foreach ((string id, JsonNode employee) in employees)
+            {
+                Assert.NotEqual(prevId, id);
+                prevId = id;
+
+                Assert.IsType<JsonObject>(employee);
+            }
+        }
+
+        [Fact]
+        public static void TestAddingKeyValuePairsCollectionAfterInitialization()
+        {
+            var employees = new JsonObject();
+            employees.AddRange(EmployeesDatabase.GetTenBestEmployees());
+
+            string prevId = "";
+            foreach ((string id, JsonNode employee) in employees)
+            {
+                Assert.NotEqual(prevId, id);
+                prevId = id;
+
+                Assert.IsType<JsonObject>(employee);
+            }
+        }
+
+        [Fact]
+        public static void TestContains()
+        {
+            var person = new JsonObject
+            {
+                { "name", "John" },
+                { "ssn", "123456789" },
+            };
+
+            if (person.ContainsProperty("ssn"))
+            {
+                EmployeesDatabase.CheckSSN(((JsonString)person["ssn"]).Value);
```

Comment:
Scenario-based testing is interesting, but this doesn't seem to be adding any value. Would it be simpler (and more meaningful) to just assert the value is 123456789?